scripts/checkStorageLayout.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+CONTRACTS=("Arbitrum_SpokePool" "Optimism_SpokePool" "Polygon_SpokePool" "Linea_SpokePool" "ZkSync_SpokePool" "Ethereum_SpokePool" "Base_SpokePool")
+if [[ "$1" == "--overwrite" ]]; then
+    for CONTRACT in "${CONTRACTS[@]}"; do
+        echo "Overwrite flag detected. Creating new storage layout snapshot of the $CONTRACT contract"
+        forge inspect $CONTRACT storagelayout > ./storage-layouts/$CONTRACT.json
+        echo "✅ 'forge inspect' saved new $CONTRACT storage layout at './storage-layouts/$CONTRACT.json'."
+    done
+    exit 0
+fi
+
+for CONTRACT in "${CONTRACTS[@]}"; do
+    echo "Comparing storage layout snapshot of the $CONTRACT contract at ./storage-layouts/$CONTRACT.json with current storage layout"
+    touch ./storage-layouts/proposed.$CONTRACT.json
+    echo "Created temporary storage layout file at ./storage-layouts/proposed.$CONTRACT.json"
+    forge inspect $CONTRACT storagelayout > ./storage-layouts/proposed.$CONTRACT.json
+    echo "'forge inspect' command created temp. storage lay out file!"
+    ## TODO: add an automatic check here like, if there is diff, throw error. This forces
+    ## the developer to include the updated JSON file in the storage-layouts/ directory
+    ## in their commit, assuming this script is running in CI.
```

Comment:
Wait, isn't this what the if statement below does? Or am I misunderstanding this TODO?

Reply:
yeah it does this is left over